crates/noirc_evaluator/src/ssa_refactor/ir/dom.rs

```diff
@@ -0,0 +1,423 @@
+        let mut func = Function::new("func".into(), func_id);
+        let block0_id = func.entry_block();
+        func.dfg.set_block_terminator(
+            block0_id,
+            TerminatorInstruction::Return { return_values: vec![] },
+        );
+        let cfg = ControlFlowGraph::with_function(&func);
+        let post_order = PostOrder::with_function(&func);
+        let dom_tree = DominatorTree::with_cfg_and_post_order(&cfg, &post_order);
+        assert_eq!(dom_tree.dominates(block0_id, block0_id), true);
+    }
+
+    // Testing setup for a function with an unreachable block2
+    fn unreachable_node_setup(
+    ) -> (DominatorTree, BasicBlockId, BasicBlockId, BasicBlockId, BasicBlockId) {
+        // func() {
+        //   block0(cond: u1):
+        //     jmpif v0 block2() block3()
+        //   block1():
+        //     jmp block2()
+        //   block2():
+        //     jmp block3()
+        //   block3():
+        //     return ()
+        // }
+        let func_id = Id::test_new(0);
+        let mut builder = FunctionBuilder::new("func".into(), func_id);
+
+        let cond = builder.add_parameter(Type::unsigned(1));
+        let block1_id = builder.insert_block();
+        let block2_id = builder.insert_block();
+        let block3_id = builder.insert_block();
+
+        builder.terminate_with_jmpif(cond, block2_id, block3_id);
+        builder.switch_to_block(block1_id);
+        builder.terminate_with_jmp(block2_id, vec![]);
+        builder.switch_to_block(block2_id);
+        builder.terminate_with_jmp(block3_id, vec![]);
+        builder.switch_to_block(block3_id);
+        builder.terminate_with_return(vec![]);
+
+        let ssa = builder.finish();
+        let func = ssa.functions.first().unwrap();
+        let block0_id = func.entry_block();
+
+        let cfg = ControlFlowGraph::with_function(&func);
+        let post_order = PostOrder::with_function(&func);
+        let dt = DominatorTree::with_cfg_and_post_order(&cfg, &post_order);
+        (dt, block0_id, block1_id, block2_id, block3_id)
+    }
+
+    // Expected dominator tree
+    // block0 {
+    //   block2
+    //   block3
+    // }
+
+    // Dominance matrix
+    // ✓: Row item dominates column item
+    // !: Querying row item's dominance of column item panics. (i.e. invalid)
+    //    b0  b1  b2  b3
+    // b0 ✓   !   ✓   ✓
+    // b1 !   !   !   !
+    // b2     !   ✓
+    // b3     !       ✓
+    // Note that from a local view block 1 dominates blocks 1,2 & 3, but since this block is
+    // unreachable, performing this query indicates an internal compiler error.
+    #[test]
+    fn unreachable_node_asserts() {
+        let (dt, b0, _b1, b2, b3) = unreachable_node_setup();
+
+        assert_eq!(dt.dominates(b0, b0), true);
+        assert_eq!(dt.dominates(b0, b2), true);
+        assert_eq!(dt.dominates(b0, b3), true);
+
+        assert_eq!(dt.dominates(b2, b0), false);
+        assert_eq!(dt.dominates(b2, b2), true);
```

Comment:
```suggestion
        assert!(!dt.dominates(b2, b0));
        assert!(dt.dominates(b2, b2));
```

All these assertions are throwing clippy errors. (and elsewhere in the file)

Reply:
That's weird, I'm clippy isn't picking this up. Should I be using any flags?
```
aztec@Aztecs-MBP noirc_evaluator % cargo clippy --version
clippy 0.1.66 (69f9c33d 2022-12-12)
```